pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -707,11 +719,135 @@ def phase_str(phi: float) -> str:
             if ax_lims[i][0] < 0:
                 ax.axhline(0, **hline_kwargs)
 
+    return (fig_reg if register else None, fig, ch_axes, data)
+
+
+def draw_samples(
+    sampled_seq: SequenceSamples,
+    register: Optional[BaseRegister] = None,
+    sampling_rate: Optional[float] = None,
+    draw_phase_area: bool = False,
+    draw_phase_shifts: bool = False,
+    draw_phase_curve: bool = False,
+) -> tuple[Figure | None, Figure]:
+    """Draws a SequenceSamples.
+
+    Args:
+        sampled_seq: The input samples of a sequence of operations.
+        register: If present, draw the register before the pulse
+            sequence samples, with a visual indication (square halo)
+            around the qubits masked by the SLM.
+        sampling_rate: Sampling rate of the effective pulse used by
+            the solver. If present, plots the effective pulse alongside the
+            input pulse.
+        draw_phase_area: Whether phase and area values need to be shown
+            as text on the plot, defaults to False. If `draw_phase_curve=True`,
+            phase values are ommited.
+        draw_phase_shifts: Whether phase shift and reference information
+            should be added to the plot, defaults to False.
+        draw_phase_curve: Draws the changes in phase in its own curve (ignored
+            if the phase doesn't change throughout the channel).
+    """
+    slot_tfs = [
+        ch_samples.slots[-1].tf
+        for ch_samples in sampled_seq.channel_samples.values()
+    ]
+    max_slot_tf = max(slot_tfs) if len(slot_tfs) > 0 else None
```

Comment:
This is introduced since if the total_duration is less that the last slot.tf, it raise an out of bound while accessing .amp, .phase and .det